### PR TITLE
Add ServiceError class

### PR DESCRIPTION
**Pull request by essame**
What does this PR do?
----------------------
Add new error class to be used when throwing errors from services to bff/abff.

### CODE REVIEW

File: src/general.js [hunk 1]
```diff
@@ -1,3 +1,11 @@
+export class ServiceError extends Error {
```
Comment (faizhasim) [inline src/general.js:1]: How is this error going to be used?

Does it mean that the service that get invoked by BFF will give an indication on what kind of http status code to BFF?

I'm wondering if it's better to just return [Boom](https://github.com/hapijs/boom/) error, rather than introducing new error here? Reasons:

- BFF already use that, so it's not something new to us.
- Since lambda invocation does not preserve JS types anyway, we can't really use the exported `ServiceError` type. Maybe it's unnecessary to export this.
- Boom error can be immediately stringified. So, I think the lambda can just reject with something like `JSON.stringify(Boom.notFound('job not found'))`. So, BFF can `JSON.parse(lambdaResult)`, introspect whether this is Boom error, then just propagate the status code response as needed.
- Bonus: Because Boom error can be properly `JSON.stringify`, this will make error logging easier.

image

Comment (essame) [reply src/general.js:1]: My intention was to use it here https://github.com/seekinternational/cpp-admin-backend-for-frontend/blob/master/src/middleware/errorHandler.js#L12.

- Boom is only used in BFF. I was a bit hesitant to introduce something like Boom which is HTTP related stuff to the services, so I went with creating a simpler class instead.
- We need to export it to use be able to throw it from the service code.
- Yea BFF/ABFF will need to check/parse the error whether it is Boom or ServiceError, it feels it is the same thing.
- Basically we only care about the error message and that is logged anyway when we throw errors from services.

You mentioned a good point that error thrown from service to ABFF/BFF will lose its type, after reading this again https://github.com/seekinternational/cpp-helpers/blob/master/src/invokeLambdaHandler.js I am guessing whether it is Boom error or ServiceError it is not going to work.
